core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -337,7 +337,7 @@ public void updateRoutingEntry(final ShardRouting newRouting, final boolean pers
                 }
                 // if its the same routing except for some metadata info, return
                 if (currentRouting.equalsIgnoringMetaData(newRouting)) {
```

Comment:
I replaced the check by an equals (no need to ignore metadata now). The check is still needed as we only want to call the shardRoutingChanged listener later if something changed.